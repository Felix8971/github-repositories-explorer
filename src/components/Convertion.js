import React from 'react';
import {updateInputTextAction} from '../actions';
import {connect} from 'react-redux';
import {transformInputSelector} from '../Selectors';

/* input
    { 
    "0": [ 
           { "id": 10, "title": "House", "level": 0, "children": [], "parent_id": null} 
         ],
    "1": [ 
           { "id": 12, "title": "Red Roof", "level": 1, "children": [], "parent_id": 10},
           { "id": 18, "title": "Blue Roof","level": 1, "children": [], "parent_id": 10},
           { "id": 13, "title": "Wall",     "level": 1, "children": [], "parent_id": 10}
         ],
    "2": [ 
           { "id": 17, "title": "Blue Window", "level": 2, "children": [], "parent_id": 12}, 
           { "id": 16, "title": "Door", "level": 2, "children": [], "parent_id": 13 },
           { "id": 15, "title": "Red Window", "level": 2, "children": [], "parent_id": 12 }
         ]
    }

    we will insert 12, 18 and 13 inside the children property of 10 because they have "parent_id": 10
    we will insert 17 and 15 inside the children array of 12
    we will insert 16 inside the children array of 13
   
  */

/*
 In this component, we have two exports.
 The first export will export the dumb component Convertion and the default export will export the connected component connect(Convertion).
 We export both so that if we want to :
  * Test just the React component alone, we test the dumb component.
  * Test React-Redux part of the component, then test the connected component.
*/

export class Convertion extends React.Component {
  constructor(props) {
    super();
    this.handleInput = this.handleInputChange.bind(this);
  }
  
  handleInputChange(event, onInputChange) {
    onInputChange(event.target.value);
    // console.log(event.target.value);
  }

  render() {
    const self = this;
    // console.log('this.props=',this.props);
    return (
      <div className="containerConvertion">
        <div className="column">
          <div>Input</div>
          <textarea 
            rows="4"
            cols="50"
            className="input"
            onChange={event => this.handleInputChange(event, self.props.onInputChange)}
            value={this.props.input}
            placeholder="Copy past a json file here..."
          >
          </textarea>
        </div>
        <div className="column">
          <div>Output</div>   
          <textarea rows="4" cols="50" className="output" value={this.props.output} onChange={() => {}} >
            <pre></pre>
          </textarea>
        </div>
      </div>
    );
  
  }
}

const mapStateToProps = (state) => {
  return {
    input: state.inputText, 
    output: transformInputSelector(state.inputText), 
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    onInputChange : (data) => {
      dispatch(updateInputTextAction(data));
    },
  };
}

//Connects the App component to the Redux store.
export default connect(mapStateToProps, mapDispatchToProps)(Convertion);
