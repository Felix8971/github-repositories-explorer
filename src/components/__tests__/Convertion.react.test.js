import React from 'react';
import { shallow /*, mount, render*/ } from 'enzyme';
import {Convertion} from '../Convertion';
import * as enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
enzyme.configure({ adapter: new Adapter() });

describe('Convertion', () => {

  it('should be defined', () => {
    expect(Convertion).toBeDefined();
  });

  it('should render correctly', () => {
    const tree = shallow(
      <Convertion/>
    );
    expect(tree).toMatchSnapshot();
  });

  it('should contain 2 textarea', () => {
    const tree = shallow(
      <Convertion/>
    );
    expect(tree.find('textarea')).toHaveLength(2);
  })
});