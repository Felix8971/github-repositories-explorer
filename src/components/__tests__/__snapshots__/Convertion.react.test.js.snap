// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Convertion should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Convertion />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="column"
        >
          <div>
            Input
          </div>
          <textarea
            className="input"
            cols="50"
            onChange={[Function]}
            placeholder="Copy past a json file here..."
            rows="4"
          />
        </div>,
        <div
          className="column"
        >
          <div>
            Output
          </div>
          <textarea
            className="output"
            cols="50"
            onChange={[Function]}
            rows="4"
          >
            <pre />
          </textarea>
        </div>,
      ],
      "className": "containerConvertion",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Input
            </div>,
            <textarea
              className="input"
              cols="50"
              onChange={[Function]}
              placeholder="Copy past a json file here..."
              rows="4"
            />,
          ],
          "className": "column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Input",
            },
            "ref": null,
            "rendered": "Input",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "input",
              "cols": "50",
              "onChange": [Function],
              "placeholder": "Copy past a json file here...",
              "rows": "4",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "textarea",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div>
              Output
            </div>,
            <textarea
              className="output"
              cols="50"
              onChange={[Function]}
              rows="4"
            >
              <pre />
            </textarea>,
          ],
          "className": "column",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Output",
            },
            "ref": null,
            "rendered": "Output",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <pre />,
              "className": "output",
              "cols": "50",
              "onChange": [Function],
              "rows": "4",
              "value": undefined,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "pre",
            },
            "type": "textarea",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="column"
          >
            <div>
              Input
            </div>
            <textarea
              className="input"
              cols="50"
              onChange={[Function]}
              placeholder="Copy past a json file here..."
              rows="4"
            />
          </div>,
          <div
            className="column"
          >
            <div>
              Output
            </div>
            <textarea
              className="output"
              cols="50"
              onChange={[Function]}
              rows="4"
            >
              <pre />
            </textarea>
          </div>,
        ],
        "className": "containerConvertion",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Input
              </div>,
              <textarea
                className="input"
                cols="50"
                onChange={[Function]}
                placeholder="Copy past a json file here..."
                rows="4"
              />,
            ],
            "className": "column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Input",
              },
              "ref": null,
              "rendered": "Input",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "input",
                "cols": "50",
                "onChange": [Function],
                "placeholder": "Copy past a json file here...",
                "rows": "4",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "textarea",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div>
                Output
              </div>,
              <textarea
                className="output"
                cols="50"
                onChange={[Function]}
                rows="4"
              >
                <pre />
              </textarea>,
            ],
            "className": "column",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Output",
              },
              "ref": null,
              "rendered": "Output",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <pre />,
                "className": "output",
                "cols": "50",
                "onChange": [Function],
                "rows": "4",
                "value": undefined,
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "pre",
              },
              "type": "textarea",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
